docs/useMethods.md

```diff
@@ -0,0 +1,48 @@
+# `useMethods`
+
+React hook that simplifies the `useReducer` implementation.
+
+## Usage
+
+```jsx
+import { useMethods } from 'react-use';
+
+const initialState = {
+  count: 0,
+};
+
+function createMethods(state) {
+  return {
+    reset() {
+      return initialState;
+    },
+    increment() {
+      return { ...state, count: state.count + 1 };
+    },
+    decrement() {
+      return { ...state, count: state.count - 1 };
+    },
+  };
+}
```

Comment:
What do you think about providing `state` to every method?

```ts
const methods = {
  reset: state => () => initialState,
  increment: state => () => { ...state, count: state.count + 1 },
  decrement: state => () => { ...state, count: state.count - 1 },
};
```

Reply:
That is an interesting idea but I feel that the current design looks better since the methods will have closure over `state` anyway. Otherwise, the definition of `methods` might be tricky to get heads around. Right now, it is just a function that returns some methods. Thoughts?

Reply:
I'm fine with either approach. Just wanted to hear your thoughts, because the current implementation on every state change has to generate a new object `{ reset, increment, ... }`, whereas in the other implementation only one function is generated.